### PR TITLE
fix: add col width, fix table width error when `selections === true && type === 'checkbox'`

### LINKED ISSUE #40814: [table] rowSelection.selections 有值时，选择列列宽不够
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/ke-xuan-ze-antd-5-2-2-forked-mwquv0?file=/demo.tsx)

### Steps to reproduce

查看渲染结果

### What is expected?

首列表头的宽度增大到 64px, 以便下拉箭头和表格线框有足够的空间

### What is actually happening?

下拉箭头和表格线框贴在了一起

| Environment | Info |
| --- | --- |
| antd | 5.2.2 |
| React | All |
| System | All |
| Browser | All |

---
虽然可以用  rowSelection.columnWidth 来自己改，但这个逻辑应该做在内部，即：
1. 无下拉选择，默认列宽 32px
2. 有下拉选择，默认列宽 64px

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

- close #40814 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | fix: add col width, fix table width error when selections === true && type === 'checkbox' |
| 🇨🇳 Chinese | fix: 添加宽度，修复表格列宽不够的问题 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 956f48f</samp>

Fixed selection column width bug for table with multiple checkbox selections. Modified `columnWidth` property of `useSelection` hook in `components/table/hooks/useSelection.tsx` to use a default value of 64 pixels.

### <samp>🤖 Generated by Copilot at 956f48f</samp>

*  Fix selection column width inconsistency with multiple selections and checkboxes ([link](https://github.com/ant-design/ant-design/pull/41626/files?diff=unified&w=0#diff-6c5f2b8a4a1061563259571e5203e5f69eac5c1a68f2f55cf100643d59738d25L86-R88))

### CODE REVIEW

**Comment by li-jia-nan** [comment]
before:

image

after:

image

**Comment by MadCcc** [comment]
这个是不应该改样式？这个 selection 因为 absolute 所以没有撑起宽度

**Comment by li-jia-nan** [comment]
image

把 absolute 改成了 flex

**Comment by li-jia-nan** [comment]
absolute 改成 flex？

**Comment by MadCcc** [comment]
checkbox 不对齐了

**Comment by li-jia-nan** [comment]
> checkbox 不对齐了

试了一下，flex 可以做，判断 `selections === true && type === 'checkbox'` 的时候，给 checkbox 加上 `margin-left`, margin-left 的值等于下拉箭头的宽度就好了，但是感觉这样实现成本复杂起来了，不如现在这样给箭头 absolute，然后直接给 col 加宽度，毕竟我们已经暴露了 `columnWidth` 这个属性，可以直接利用起来

File: components/table/InternalTable.tsx [hunk 1]
```diff
@@ -110,11 +110,10 @@ export interface TableProps<RecordType>
   showSorterTooltip?: boolean | TooltipProps;
 }
 
-function InternalTable<RecordType extends object = any>(
+const InternalTable = <RecordType extends AnyObject = any>(
   props: InternalTableProps<RecordType>,
   ref: React.MutableRefObject<HTMLDivElement>,
-) {
-  const { getPopupContainer: getContextPopupContainer } = React.useContext(ConfigContext);
```
Comment (li-jia-nan) [inline components/table/InternalTable.tsx:117]: 下面已经有一个 useContext 了，直接合并到一起了

File: components/table/hooks/useSelection.tsx [hunk 2]
```diff
@@ -54,24 +56,24 @@ export type INTERNAL_SELECTION_ITEM =
...
+  childrenColumnName: keyof RecordType,
+  data?: RecordType[],
+): RecordType[] => {
   let list: RecordType[] = [];
   (data || []).forEach((record) => {
     list.push(record);
     if (record && typeof record === 'object' && childrenColumnName in record) {
-      list = [
-        ...list,
-        ...flattenData<RecordType>(childrenColumnName, (record as any)[childrenColumnName]),
```
Comment (li-jia-nan) [inline components/table/hooks/useSelection.tsx:64]: 为了干掉这个 `as any`，顺便优化了一下 RecordType 范型

File: components/table/hooks/useSelection.tsx [hunk 3]
```diff
@@ -684,16 +687,18 @@ function useSelection<RecordType>(
         prevCol.fixed = mergedFixed;
       }
 
+      const columnCls = classNames(`${prefixCls}-selection-col`, {
+        [`${prefixCls}-selection-col-with-dropdown`]: selections && selectionType === 'checkbox',
```
Comment (li-jia-nan) [inline components/table/hooks/useSelection.tsx:691]: 有 checkbox 并且有下拉箭头的时候添加 classname

File: components/table/hooks/useSelection.tsx [hunk 4]
```diff
@@ -410,7 +413,7 @@ function useSelection<RecordType>(
               const { key, text, onSelect: onSelectionClick } = selection;
 
               return {
-                key: key || index,
+                key: key ?? index,
```
Comment (li-jia-nan) [inline components/table/hooks/useSelection.tsx:416]: 为了兼容 0 改成 `??` 运算符

**Review by MadCcc — APPROVED** [review]

**Comment by MadCcc** [comment]
这个居然没有 argos diff

**Pull request merged**